src/System.ValueTuple/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -2060,12 +2269,52 @@ string ITupleInternal.ToStringEnd()
             }
         }
 
-        int ITupleInternal.Size
+        /// <summary>
+        /// The number of positions in this data structure.
+        /// </summary>
+        public int Length
+        {
+            get
+            {
+                ITupleInternal rest = Rest as ITupleInternal;
+                return rest == null ? 8 : 7 + rest.Length;
```

Comment:
Or rather, you'd cast to ITuple if that is the constraint.

---
In reply to: [70889577](https://github.com/dotnet/corefx/pull/10068#discussion_r70889577) [](ancestors = 70889577)